PowerValidatedSolutions.psm1

```diff
@@ -20714,6 +20721,384 @@ Function Get-PasswordPolicyConfig {
+	)
+	
+	if (($address -eq "Null") -and ($required -eq $true)) {
+		Write-LogMessage -Type ERROR -Message "$name variable has not been configured." -Colour Red
+		return $false
+	}
+	$checkStatement = $address -match "^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$"
+	if ($checkStatement -eq $true) {
+		return $true
+	} else {
+		Write-LogMessage -Type ERROR -Message "Please input a validate email address for $name " -Colour Red
+		return $false
+	}
+}
+
+Function Test-PasswordPolicyConfig {
+    Param (
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [psobject]$customConfig
+    )
+
+    # import default configuration JSON for comparing parameters
+    $defaultConfig = Get-PasswordPolicyDefault
+    $encounterError = "False"
+
+    # validating product Types in customConfig file
+    $defaultProductList = $defaultConfig | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+    $customProductList = $customConfig | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+    $defaultSection = "passwordExpiration", "passwordComplexity", "accountLockout"
+	
+    foreach ($product in $customProductList) {
+        if (-Not $defaultProductList.Name.Contains($product.Name)) {
+            Write-LogMessage -Type ERROR -Message "Unknow product type $($product.Name), please check the custom configuration file and run the cmdlet again" -Colour Red
+            $encounterError = "True"
+            break
+        }
+        # Validating product sections in customConfig file
+        if ($encounterError -ne "True") {
+            $customSectionList = $customConfig.($product.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+            foreach ($section in $customSectionList) {
+                 if (-Not $defaultSection.Contains($section.Name)) {
+                    Write-LogMessage -Type ERROR -Message "Unknow section $($section.Name) under product $($product.Name), please check the custom configuration file and run the cmdlet again" -Colour Red
+                    $encounterError = "True"
+                    break
+                }
+                # Validate parameters in customConfig file
+			    if ($encounterError -ne "True") {
+				    $defaultParameterList = $defaultConfig.($product.Name).($section.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+			    	$customParameterList = $customConfig.($product.Name).($section.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+			    	foreach ( $parameterName in $customParameterList) {
+			    		if ( -Not $defaultParameterList.Name.Contains($parameterName.Name)) {
+			    			Write-LogMessage -Type ERROR -Message "Unknow parameter $($parameterName.Name) under $($section.Name) under product $($product.Name), please check the custom configuration file and run the cmdlet again" -Colour Red
```

Comment:
Change to:

Write-LogMessage -Type ERROR -Message "Found Unknown Parameter ($($parameterName.Name)) Under Section ($($section.Name)) for Product ($($product.Name)), Please Check the Password Policy Configuration File and Run Again" -Colour Red

Reply:
resolved in 2948bc273bae34312514db8aba5d7cb584efd8d4